kord-extensions/src/main/kotlin/com/kotlindiscord/kord/extensions/ExtensibleBot.kt

```diff
@@ -192,14 +233,49 @@ open class ExtensibleBot(
      * @throws InvalidExtensionException Thrown if the extension has no primary constructor.
      */
     @Throws(InvalidExtensionException::class)
-    suspend fun addExtension(extension: KClass<out Extension>) {
+    fun addExtension(extension: KClass<out Extension>) {
         val ctor = extension.primaryConstructor ?: throw InvalidExtensionException(extension, "No primary constructor")
 
         val extensionObj = ctor.call(this)
 
         extensions[extensionObj.name] = extensionObj
     }
 
+    /**
+     * Reload an installed [Extension] from this bot, by name.
+     *
+     * This function **does not** remove the extension object - it simply
+     * removes its event handlers and commands. Unloaded extensions can
+     * be loaded again by calling [ExtensibleBot.loadExtension].
+     *
+     * @param extension The name of the [Extension] to unload.
+     */
+    @Throws(InvalidExtensionException::class)
+    suspend fun loadExtension(extension: String) {
+        val extensionObj = extensions[extension] ?: return
+
+        if (!extensionObj.loaded) {
+            extensionObj.doSetup()
+        }
+    }
+
+    /**
+     * Unload an installed [Extension] from this bot, by name.
+     *
+     * This function **does not** remove the extension object - it simply
+     * removes its event handlers and commands. Unloaded extensions can
+     * be loaded again by calling [ExtensibleBot.loadExtension].
+     *
+     * @param extension The name of the [Extension] to unload.
+     */
```

Comment:
Maybe we should note that the function just returns if no matching extension is found?